skorch/tests/callbacks/test_training.py

```diff
@@ -170,22 +170,23 @@ def epoch_3_scorer(net, *_):
         ])
         net.fit(*data)
 
-        assert save_params_mock.call_count == 3
+        assert save_params_mock.call_count == 6
         assert cb.get_formatted_files(net) == {
             'f_params': 'model_3_10.pt',
             'f_optimizer': 'optimizer_3_10.pt',
             'f_history': 'history.json',
             'f_pickle': None
         }
         save_params_mock.assert_has_calls(
-            [call(f_params='model_3_10.pt'),
+            [call(f_params='model_1_10.pt'),
+             call(f_optimizer='optimizer_1_10.pt'),
+             call(f_history='history.json'),
+             call(f_params='model_3_10.pt'),
              call(f_optimizer='optimizer_3_10.pt'),
              call(f_history='history.json')]
         )
-        assert sink.call_count == 1
-        assert all((x is False) for x in net.history[:2, 'event_cp'])
-        assert net.history[2, 'event_cp'] is True
-        assert all((x is False) for x in net.history[3:, 'event_cp'])
+        assert sink.call_count == 2
+        assert [True, False, True] + [False] * 7 == net.history[:, 'event_cp']
```

Comment:
I think it would be beneficial to have a short explanation why the result looks like it does (what makes the result True in those cases?).